### DISCUSSION TITLE
Pygetpapers structure discussion

### DISCUSSION THREAD

**Discussion by ayush4921** [petermr/dictionary#22] [Ideas]
Current Status: Have got the initial set of requirements and need some community support
The problem: I now know how to access the EPMC repo using apis but need some insight onto how I should model the project.
Specifically which format of the command of the pygetpapers will the users prefer
Currently we have two main options:
     1.Add codes for getting all the information as methods inside a subclass which will represent a repository which is inside the main 
        pygetpapers class
     2. We can add parameters to the main pygetapi itself which will take in the repo as well as the information that the user wants (edited) 
         I dont want to add the code in a non maintainable format therefore am not coding any classes right now. I am making functions 
         which I will later model according to our project structure
Giving insight into what the default parameters for the api will also be useful
Currently I have set relevency as the sorting and 12 as the number of papers
But we can change that if we add a parameter for them

Please Remove the requirements that should not be the part of this specific segment. As this is the process of getting the paper. We will use other tools to do processing on them.

Please Add requirements which you feel should be a part of the mechanism that gets us the paper. An example of it would be to download all the introductions from papers regarding a query

The link for requirements is available at: https://github.com/petermr/openVirus/wiki/pygetpapers

The code is available at https://github.com/petermr/dictionary/blob/main/pygetpapersdev/europepmc.py

This is not the final code but just a representation of how we can access the europepmc api in python.
Please comment on the structure and we can then model the code according to the requirements decided.

**Comment by petermr** [comment]
Please clarify what the issue is:
* the title should be meaningful, e.g "architecture of `pygetpapers`"
* code should be quoted / formatted - need to know where code starts and text ends

But we need a high-level architecture diagram, with components such as query (creator), searcher and results . 
We need to know what each of these does.
